src/librustc/diagnostics.rs

```diff
@@ -360,6 +360,39 @@ integer type:
 http://doc.rust-lang.org/reference.html#ffi-attributes
 "##,
 
+E0109: r##"
+You tried to give type parameter to a type which doesn't need it. Erroneous
+code example:
+
+```
+type X = u32<i32>; // error: type parameters are not allowed on this type
+```
+
+Please check you actually used the good type or check again its definition.
```

Comment:
"Please check that you used the correct type and recheck its definition. Perhaps it doesn't need the type parameter."